tasks/setup.py

```diff
@@ -31,16 +32,34 @@ def setup(
     else:
         info("🤖 Install Pulumi")
         if is_windows():
-            os.system("winget install pulumi")
+            ctx.run("winget install pulumi")
         elif is_linux():
-            os.system("curl -fsSL https://get.pulumi.com | sh")
+            ctx.run("curl -fsSL https://get.pulumi.com | sh")
         else:
-            os.system("brew install pulumi/tap/pulumi")
+            ctx.run("brew install pulumi/tap/pulumi")
+        if shutil.which("pulumi") is None:
+            print()
+            warn("Pulumi is not in the PATH, please add pulumi to PATH before running tests")
+            # Pulumi is not in the PATH, add it to the process env so rest of setup can continue
+            if is_windows():
+                # Add common pulumi install locations to PATH
+                paths = [
+                    str(x)
+                    for x in [
+                        Path().home().joinpath(".pulumi", "bin"),
+                        Path().home().joinpath("AppData", "Local", "pulumi", "bin"),
+                        'C:\\Program Files (x86)\\Pulumi\\bin',
+                    ]
+                ]
+                os.environ["PATH"] = ';'.join([os.environ["PATH"]] + paths)
+            else:
+                path = Path().home().joinpath(".pulumi", "bin")
```

Comment:
Why would that be necessary?

Reply:
If the above lines install pulumi for he first time, then it's not on the path for the process because it was only just installed. This means that the `pulumi plugin install` and `pulumi login --local` commands fail. This was hidden before because `os.system` continues on error.